src/rinterface.c

```diff
@@ -7982,11 +7982,17 @@ SEXP R_igraph_hrg_sample(SEXP hrg) {
 
   SEXP r_result;
                                         /* Convert input */
-  R_SEXP_to_hrg(hrg, &c_hrg);
+  if (0 != R_SEXP_to_hrg_copy(hrg, &c_hrg)) {
+    igraph_error("", __FILE__, __LINE__, IGRAPH_ENOMEM);
```

Comment:
In the Stimulus template:

```suggestion
    igraph_error("internal error in R_SEXP_to_hrg_copy()", __FILE__, __LINE__, IGRAPH_ENOMEM);
```